### ISSUE TITLE
How to send CECHOResponse, After Receiving a CECHOREQUEST

### ISSUE THREAD

**Issue by C-Sagar-Kumar** [fo-dicom/fo-dicom-samples#75]
Hi,
I am using below code to send the CECHOResponse
 but Its timing out, can you guide me what exactly I am missing in my below code.

 Dicom.Network.Client.DicomClient dicomClient = CreateDicomVerificationClient(strIPAddress, port);                           
 await dicomClient.SendAsync();

![code](https://user-images.githubusercontent.com/30107495/142445990-6b468c71-eab6-4810-bd0a-6d35419b856c.PNG)
![log](https://user-images.githubusercontent.com/30107495/142446479-a74ed260-9b3f-419a-9169-9ed7dedaff20.PNG)

**Comment by gofal** [comment]
you called "client.AddRequestAsync(dicomCEchoRequest);" without the "await". that means maybe the request is not ready added to the pipeline when you started sending the request.

**Comment by C-Sagar-Kumar** [comment]
Hi Reinhard, Thnx for you reply. I fixed the issue long back, actual issue was with the simulator which I was using for verification call testing is actually the SCP and I need to create and send as SCU. However, I tried as SCP only so its not able to connect to the SCU and sent the response as time out. I found out and resolved this and also what feedback you gave that also I found out and fixed since it was async call without await it will run synchronously and execute without waiting for completion of the thread. I forgot to update this communication thread after my fix sorry for that.
Thanks for your reply and Have a great day.

**Comment by gofal** [comment]
thanks for your answer and how you fixed it.

**Comment by C-Sagar-Kumar** [comment]
As I told Since the simulator(which is the user/ ultrasound machine for me for testing) was the SCU, I created the SCP provider i.e. Created the custom class and implemented & inherited  DicomService, IDicomServiceProvider, IDicomCStoreProvider, IDicomCEchoProvider, IDicomCFindProvider and then wrote the necessary logic as per my requirement then it resolved the issue. 
Lemme know If I can create a sample project for the same and upload into the fo-dicom-samples repo, so that it will be easy to implement for others, bcoz I faced so many issues while implementing the same since the solution/ samples we got from google is not straight forward. Therefore, I broke my head so much to get exact root cause and solution.
